src/WebJobs.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -143,7 +145,33 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                     if (Enum.TryParse((string)kindValue, out TokenSourceType tokenSourceKind) &&
                         tokenSourceKind == TokenSourceType.AzureManagedIdentity)
                     {
-                        return new ManagedIdentityTokenSource((string)jsonObject.GetValue("resource", StringComparison.Ordinal));
+                        string resourceString = (string)jsonObject.GetValue("resource", StringComparison.Ordinal);
```

Comment:
Are these changes really necessary? It seems like your custom `ManagedIdentityOptions` should serialize/deserialize correctly already?

Reply:
These changes are there to correctly serialize/deserialize the json given from out of process requests. I followed the json format from the tests in OutOfProcTests.cs (shown below). Let me know if there's a better way to format the `options` in the out of process request to avoid these changes.

```
var executionJson = @"
{
    ""isDone"": false,
    ""actions"": [
        [{
            ""actionType"": ""CallHttp"",
            ""httpRequest"": {
                ""method"": ""GET"",
                ""uri"": ""https://example.com"",
                ""tokenSource"": {
                    ""kind"": ""AzureManagedIdentity"",
                    ""resource"": ""https://management.core.windows.net"",
                    ""options"": {
                        ""authorityhost"": ""https://login.microsoftonline.com/"",
                        ""tenantid"": ""example_tenant_id""
                    }
                }
            }
        }]
    ]
}";
```